src/scripts/main.js

```diff
@@ -1,3 +1,41 @@
 'use strict';
 
-// write code here
+const listItems = [...document.querySelectorAll('li')];
+const employeesList = document.querySelector('ul');
+
+function convertToNumber(string) {
+  const salaryNumber = string.replace('$', '').replace(',', '');
+
+  return +salaryNumber;
+}
+
+function sortBySalary(items) {
+  items.sort(
+    (a, b) =>
+      convertToNumber(b.dataset.salary) - convertToNumber(a.dataset.salary),
+  );
+}
+
+function updateList(list, items) {
+  list.innerHTML = '';
+  items.forEach((item) => list.appendChild(item));
+}
```

Comment:
In this function, you're clearing the list's innerHTML and then re-adding each item. This could be inefficient if the list is large. Instead, you could create a document fragment, append the items to it, and then append the fragment to the list. This would result in only one reflow and repaint operation, improving performance.